mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -126,6 +126,10 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
         Send whatever message is currently in the chat input
     */
     const sendChatInputMessage = async (input: string) => {
+        // Step 0: If code diffs are displayed, reject the AI code, then send the message.
+        if (unifiedDiffLines !== undefined) {
+            rejectAICode()
+        }
```

Comment:
Okay, so now seeing this in action: In a lot of ways this is better because it stops users from having broken code in their notebook (ie: the diffed code that is a mash up of the original code and the AI generated code)... but there is also a big bug here.  In particular, it opens up a dangerous flow that you can see here. 

1. Make an AI edit and see the response. 
2. Then: A user continue to assume that their edit is applied and there is no more actions that they need to do to accept it (which is what we're seeing users do), so they move to the next code cell to make their next edit
3. They use the AI to prompt for a new update. 
4. The `rejectAICode` writes the original code to the correct active code cell, overwriting what was in the code cell. 

https://github.com/user-attachments/assets/23e4dc91-6ed4-42c6-a329-2f829279c6ea

This is now more likely to happen because before users had to hit the `reject` button to deny the changes. Now, they just have to keep working. 

Here's a proposal on how to fix this: 

1. Create a new type called 

```python
export interface IAIOptimizedChatHistory {
    message: OpenAI.Chat.ChatCompletionMessageParam
    codeCellID: str
}
```

2. When we add a new message to the chat history manager, get the active code cell id and add it to the chat. We can do this very similarly to how we are currently getting the correct active code cell content. 

3. When we call the `rejectAICode` button, instead of overwriting the code in the current active code cell, overwrite the code in the code cell id. 

While we're at it, we can also update it so when the user accepts an AI response, we also write the code based on the saved code cell id instead of the current active code cell.